packages/shared/lib/services/providers.ts

```diff
@@ -42,22 +42,29 @@ export async function monitorProviders(): Promise<() => void> {
     providers = JSON.parse(providersRaw) as Record<string, Provider>;
     logger.info(`Providers loaded from url ${providersUrl} (${providersHash})`);
 
-    const timeout = setInterval(async () => {
-        try {
-            const providersRaw = await fetchProvidersRaw(providersUrl);
-            const newProvidersHash = createHash('sha1').update(providersRaw).digest('hex');
-
-            if (newProvidersHash !== providersHash) {
-                providersHash = newProvidersHash;
-                providers = JSON.parse(providersRaw) as Record<string, Provider>;
-                logger.info(`Providers reloaded (${providersHash})`);
+    let running = true;
+    void (async function () {
+        while (running) {
+            await new Promise((resolve) => setTimeout(resolve, reloadInterval));
```

Comment:
import { setTimeout } from node:timers  can help

Reply:
Did that in dbcd7164. Also I didn't realize that was a thing. Thanks!